code/modules/spatial_grid/dummy_source.dm

```diff
@@ -0,0 +1,51 @@
+/// Global list of dummy atoms in the form [z][x][y] scaled using GET_SPATIAL_INDEX
+/// They sit in the bottom left corner of spatial grid cells and act as owners for things that want to be detached (mostly datums)
+GLOBAL_LIST_EMPTY_TYPED(dummy_grid_atoms, /obj/effect/abstract/dummy_grid_source)
+
+/proc/update_dummy_sources(z_level)
+	var/list/layer = list()
+	for(var/x in 1 to SPATIAL_GRID_CELLS_PER_SIDE(world.maxx))
+		var/list/row = list()
+		for(var/y in 1 to SPATIAL_GRID_CELLS_PER_SIDE(world.maxy))
+			var/turf/bottom_left = locate(GRID_INDEX_TO_COORDS(x), GRID_INDEX_TO_COORDS(y), z_level)
+			row += new /obj/effect/abstract/dummy_grid_source(bottom_left)
+		layer += list(row)
+	GLOB.dummy_grid_atoms += list(layer)
+
+/obj/effect/abstract/dummy_grid_source
```

Comment:
i dont think abstract objects are necessary or desired, itd be better to have the datum handle its own membership inside the grid